server/eclipse-project/src/main/java/us/freeandfair/corla/controller/BallotSelection.java

```diff
@@ -134,23 +135,25 @@ public void addCvrIds(final List<Long> cvrIds) {
     }
 
     /**
-     * Return a list of CVR IDs deduped and sorted. To be given to an
-     * audit board.
+     * Return a list of CVRs, de-duplicated and sorted.
+     *
+     * The IDs of these CVRs are expected to be given to an audit board.
+     *
+     * PERF: This is not a fast operation, avoid calling it more than once.
      */
-    public List<Long> ballotSequence() {
-      return cvrs.stream()
-        .map(cvr -> cvr.id())
-        .collect(Collectors.toList());
+    public List<CastVoteRecord> cvrsInBallotSequence() {
+      return BallotSequencer.sortAndDeduplicateCVRs(
+          cvrs.stream().collect(Collectors.toList()));
```

Comment:
Expensive operation, but downstream consumers won't call this more than once per county dashboard.